schema/dataKeys.json

```diff
@@ -560,17 +564,186 @@
         }
       },
       "then": {
-        "required": [
-          "cdn"
-        ],
+        "required": ["cdn"],
         "properties": {
           "cdn": {
+            "required": ["env"]
+          }
+        }
+      }
+    },
+    {
+      "if": {
+        "properties": {
+          "systems": {
+            "contains": {
+              "const": "fbc"
+            }
+          }
+        }
+      },
+      "then": {
+        "required": ["fbc"],
+        "properties": {
+          "fbc": {
             "required": [
-              "env"
+              "allowedPackages",
+              "fromIndex",
+              "targetIndex",
+              "publishingCredentials",
+              "buildTimeoutSeconds",
+              "issueId",
```

Comment:
we have 3 types of FBC release: normal, preGa and hotfix. Some are common, other are only for preGA and other for hotfix. If I understand correctly, would this make all FBC require these keys?

Reply:
So this would make all 3 them required

Reply:
Yes. Unless there is a way to create some conditions, like if hotfix, then... if preGA, then... otherwise, we should make only the common tags required. Also, I forgot to mention that when "stagedIndex" is true, the targetIndex should be empty :D

Reply:
The following is true:

Required *common* parameters:
```
      fromIndex
      requestUpdateTimeout 
      buildTimeoutSeconds
      publishingCredentials
      allowedPackages
```
required for *prod*  builds
```
all common
targetIndex
```
required for *stagedIndex* builds
```
all common
stagedIndex

``` 
required for *hotfix* builds
```
all common
targetIndex
hotfix
issueId

```
 required for *preGA* builds
```
all common
preGA
productName
productVersion
```

Reply:
I updated the comment as the targetIndex was missing for preGA.

Reply:
I think this is fine. The two you mentioned, they functionality rely on having these fields set. In the other hand, the stagedIndex will work the same way having the targetIndex or not.

Reply:
Ah, the prod build means that stagedIndex is not set. Maybe that; s the only thing that still worth it to check. If `stageIndex` is false, the targetIndex is required.

Reply:
Fixed